packages/scene-composer/src/interfaces/dataTypes.ts

```diff
@@ -34,6 +34,11 @@ export interface ITransformConstraint {
 }
 
 export type { IValueDataBinding, ITwinMakerEntityDataBindingContext } from '@iot-app-kit/source-iottwinmaker';
+export type TargetMetadata<T> = Record<string, T> & {
```

Comment:
maybe remove `Record<string, T> &`  part and make this type more specific, so we have better typing support to avoid typos. 
I've seen problem with several other fields typed as `Record<string, xxx>` that we can hardly know the values being stored. so I think we should start to type with the clear keys and extend in the future when we need more keys, instead of starting with a generic map type

Reply:
In our case T is color,icon and iconName and in the future if you all together want to create TargetMetadata of different type then this record still holds good. I guess we can keep as it is but internally color, icon and iconName we can specify a type. I would like to keep this generic type as it is but T type can be specified internally.

Reply:
I don't think a set of metadata will always have the same type, so the generic type here will not be super useful here. and also if we need more kinds of metadata, we should defined in here to make it more clear. so the generic `Record<>` part will not be useful as well

Reply:
You meant to they will have the same type? If it is not of same then, T holds good. All you have to do is 
```
color?: string;
 iconPrefix?: string;
 iconName?: string;
opacity?: Object;
meta?: string[];
```
If we string just to Record<string, string>, the above case does not work. That time again we will have to create a new type.

Reply:
so whenever you add new metadata, you will add the field explicitly here. there isn't a case we will use `Record<string, T>`. it just make us unsure about the exact metadata can be in the metadata

Reply:
I've recently changed some types defined as `Record<xx, yy>` to be explicit interfaces, because I'm seeing problem for us to know the data stored without searching in the entire code base